src/main/java/de/numcodex/feasibility_gui_backend/query/QueryHandlerRestController.java

```diff
@@ -68,4 +86,147 @@ public ResponseEntity<Object> runQuery(
+      }
+    }
+    try {
+      var query = queryHandlerService.getQuery(queryId, authorId);
+      StoredQuery storedQuery = StoredQueryConverter.convertPersistenceToApi(query);
+      List<TermCode> invalidTermCodes = termCodeValidation.getInvalidTermCodes(storedQuery);
+      storedQuery.setInvalidTerms(invalidTermCodes);
+      return new ResponseEntity<>(storedQuery, HttpStatus.OK);
+    } catch (JsonProcessingException e) {
+      return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+    } catch (IllegalAccessException e) {
+      return new ResponseEntity<>(HttpStatus.FORBIDDEN);
+    }
+  }
+
+  @GetMapping(path = "/stored-query")
+  public ResponseEntity<Object> getStoredQueryList(@RequestHeader(HttpHeaders.AUTHORIZATION) String authorizationHeader, Principal principal) {
+    String authorId;
+
+    if (keycloakEnabled) {
+      authorId = principal.getName();
+    } else {
+      try {
+        authorId = getUserIdFromAuthorizationHeader(authorizationHeader);
+      } catch (IllegalAccessException e) {
+        return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+      }
+    }
+    var queries = queryHandlerService.getQueriesForAuthor(authorId);
+    var ret = new ArrayList<StoredQuery>();
+    queries.forEach(q -> {
+      try {
+        StoredQuery convertedQuery = StoredQueryConverter.convertPersistenceToApi(q);
+        convertedQuery.setStructuredQuery(null);
+        ret.add(convertedQuery);
+      } catch (JsonProcessingException e) {
+        log.error("Error converting query");
+      }
+    });
+    return new ResponseEntity<>(ret, HttpStatus.OK);
+  }
+
+  @GetMapping(path = "/stored-query/validate")
+  public ResponseEntity<Object> validateStoredQueryList(@RequestHeader(HttpHeaders.AUTHORIZATION) String authorizationHeader, Principal principal) {
+    String authorId;
+
+    if (keycloakEnabled) {
+      authorId = principal.getName();
+    } else {
+      try {
+        authorId = getUserIdFromAuthorizationHeader(authorizationHeader);
+      } catch (IllegalAccessException e) {
+        return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+      }
+    }
+    var queries = queryHandlerService.getQueriesForAuthor(authorId);
+    var ret = new ArrayList<StoredQuery>();
+    queries.forEach(q -> {
+      try {
+        StoredQuery convertedQuery = StoredQueryConverter.convertPersistenceToApi(q);
+        List<TermCode> invalidTermCodes = termCodeValidation.getInvalidTermCodes(convertedQuery);
+        convertedQuery.setIsValid(invalidTermCodes.isEmpty());
+        convertedQuery.setStructuredQuery(null);
+        ret.add(convertedQuery);
+      } catch (JsonProcessingException e) {
+        log.error("Error converting query");
+      }
+    });
+    return new ResponseEntity<>(ret, HttpStatus.OK);
+  }
+
+  /**
+   * Read the subject id from a Authorization header with a JWT.
+   */
+  private String getUserIdFromAuthorizationHeader(String authorizationHeader)
+      throws IllegalAccessException {
```

Comment:
This exception is used in certain Java reflection calls and should not be used for application specific exception handling.

Reply:
I removed it in [6dc7b3a](https://github.com/medizininformatik-initiative/feasibility-backend/pull/9/commits/6dc7b3a5a5d01d853a98d7f7a45a38435f79c25d)